File: app/src/main/java/nl/acidcats/teatimer/alarm/AlarmReceiver.java
package nl.acidcats.teatimer.alarm;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import nl.acidcats.teatimer.ui.TeaTimeActivity;
import timber.log.Timber;

/**
 * Receiver for alarms
 */

public class AlarmReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Timber.d("onReceive: ");

        AlarmHelper.handleAlarm(context, TeaTimeActivity.class);
    }
}

File: app/src/main/java/nl/acidcats/teatimer/alarm/AlarmReceiver.kt
package nl.acidcats.teatimer.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Receiver for alarms
 */

class AlarmReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        AlarmService.onAlarmDone(context)
    }
}
